Docs: Split and relabel VCS integration as explanation and how-to (Diátaxis) (#9675)
readthedocs/readthedocs.org: docs/user/integrations.rst

       (Perhaps reuse this: https://about.readthedocs.com/images/homepage.png)

Building documentation on every git commit
==========================================

Read the Docs is a *Continuous Documentation Deployment* platform for your software project.
Every time you change something in your documentation, Read the Docs will detect your change and build your documentation.
This happens using *webhooks*.

The Continuous Integration and Continuous Deployment (CI/CD) features are configured with your repository provider,
such as GitHub, Bitbucket or GitLab,
and with each change committed to your repository, Read the Docs is notified by the webhook.

When we receive an integration notification, we determine if the change is related to an active version for your project.
If it is, a build is triggered for that version.

Continuous Documentation for software projects
----------------------------------------------

Documentation fits into any CI/CD pipeline through the idea and objectives of *Documentation as Code*.
A straight-forward method to achieve this is by maintaining documentation code alongside the source code,

> meaning that the documentation's life-cycle is contained within the same git repository as the source code.

Author: Thanks for linking this, what a fascinating piece of inconsistency and rebellion against spelling authority :)

> Four online dictionaries I consulted (Merriam-Webster, Oxford Dictionaries, Macmillan, and Dictionary.com) all agree on the open form [life cycle](http://dictionary.reference.com/browse/life+cycle?s=t).

When changes happen to the source code, changes should also happen to the documentation.

> By managing these changes in the same life-cycle,

Author: Here, "life cycle" was actually referring back to the previous mentioning of life cycle. I like it better than "process" because it captures multiple processes combined.

So for instance to describe versioning (still missing btw!), I think the _life cycle_ of documentation is a really good term.

> By managing these changes in the same life-cycle,

Author: ```suggestion
By managing these changes in the same life cycle,
```

> you can benefit from for instance both documentation and source code being subject of the same code reviews process.

Author: Haha, yes indeed! And really good to highlight this. Readers that are developers will get this message immediately.

I'm a bit worried that readers who aren't developers will have a harder time getting it. Here, the question becomes more fundamental than using Read the Docs.. why should you even store Documentation as Code with git? A quick way to solve it would be to add some external reference about this, in case we know a great one.

When you see a "for instance", it's because my texts are generated in a flow of thoughts that are "spoken" - I try to get them massaged into a precise and concise written form afterwards :)

> Having an automated and short feedback loop for your documentation affords you to
> keep it updated and published at a low cost and effort for your contributors.

Reviewer: ```suggestion
Having an automated and short feedback loop for your documentation allows you to
keep it updated with minimal effort.
This allows more iteration on documentation,
and increases overall value from the documentation you write.
```

Continuous Documentation for all projects
-----------------------------------------

> All categories of documentation benefit from being *continuously* built and published through a git platform.

Reviewer: ```suggestion
All categories of documentation benefit from being *continuously* built and published through Read the Docs.
```

> Connecting git repositories

Reviewer: ```suggestion
Build docs on every git commit
```

I think this is something closer to what we want? 

I think this page should probably:

* Talk about the benefits of always up to date docs
* I think probably we want to discuss versioning in here, since it relies directly on Git?
* Have a small diagram that shows (You --push--> GitHub --webhook--> RTD --Build docs--> Deploy
  - (Perhaps reuse this: https://about.readthedocs.com/images/homepage.png)
 

But overall I think this is a tricky page to write as currently framed. I think we actually want something like `Benefits of Continuous Docs Deployment` or something?

Author: For the title, is "Continuous" the most important word?

"Continuous deployment with git"
"Publish or preview every commit - Continuous deployment with git"
"Git automation"

Closely aligned with this, it would be nice to have an explanation article covering "Documentation as Code" in the lead-up to talking about "Continuous".

Author: I'm adding this to a sphinx comment at the top of the article, to make its scope clear.

Reviewer: Not 100% sure what the best option is. Our doc refactor had these high-level topics. Should it be one of those? 

<img width="574" alt="Screenshot 2022-11-29 at 11 40 32 AM" src="https://user-images.githubusercontent.com/25510/204631917-ea133956-df0e-4c6d-b509-84e80e9922f0.png">

Author: Great idea - it seems best placed as a sublevel immediately under Explanation. These are necessary to click on to get to the subsublevels. So the UX will be consistent, and we won't end up with subsublevels like "Introduction" or "Overview", we can use the sublevel for that.

Author: I've added a comment about this in the planning document. We'll have to make some assessment about the sub-levels at that point.

===========================

> Read the Docs is a *Continuous Integration* platform for your git repository.

Reviewer: ```suggestion
Read the Docs is a *Continuous Documentation Deployment* platform for your software project.
```

I think this is closer to what we want to say.

Author: Sounds good :+1: The word "Continuous" is helpful in case the reader is already familiar with other build services like Travis, Jenkins, Circle.. even GitBook.. they might recognize the setup.

Every time you change something in your documentation, Read the Docs will detect your change and build your documentation.
This happens using *webhooks*.

The Continuous Integration and Continuous Deployment (CI/CD) features are configured with your repository provider,
such as GitHub, Bitbucket or GitLab,
and with each change committed to your repository, Read the Docs is notified by the webhook.

When we receive an integration notification, we determine if the change is related to an active version for your project.
If it is, a build is triggered for that version.

Continuous Documentation for software projects
----------------------------------------------

Documentation fits into any CI/CD pipeline through the idea and objectives of *Documentation as Code*.
A straight-forward method to achieve this is by maintaining documentation alongside the source code,
meaning that the documentation's life cycle is contained within the same git repository as the source code.

When changes happen to the source code, changes should also happen to the documentation.
By managing these changes in the same life-cycle,

> you can benefit from **documentation and source code being part of the same code review process**.

Author: Flagging for Tuesday: I'd like to add Versioning in quick succession after this.

> How To Integrate Your Git Platform

Reviewer 2: We do support other VCS that are not Git, so I'm not 100% sure this title is correct.

Also, the page talks about "VCS providers", so if we are changing the title, we should also rewrite those sentences as well.

Author: Aborted

If you are managing your documentation in git,
you will be able to preview changes and see accepted changes published immediately using :doc:`/pull-requests`.

Automated versioning
--------------------

Maintaining your documentation in a git repository with Read the Docs' automated CI/CD pipeline
also means that you will be able to align your project's **release cycle** with your documentation.
Whenever you announce a new **version** for your documentation,
you will probably tag that version in your git repository.

> Read the Docs automatically registers the change and adds a new version to your documentation.

Reviewer: We don't always do this, so we should not overpromise here. Users have to configure this.

Author: Yes, should definitely emphasize that it's ultimately an option that has to be configured.

Automated versioning
--------------------

With Read the Docs' automated CI/CD pipeline, you will be able to fully align your project's **release cycle** with your documentation.
For instance, a new version of a software project can build and publish a new version of its documentation.

Read the Docs is notified of all changes to your git branches and git tags.
When you release a **new version** for your project,
you are likely also adding a version tag to your git repository.

Read the Docs is notified of changes to git tags and branches.
These events can be configured to build and publish your documentation when new tags or branches are added in your repository.

> If you use a versioning schema, you can choose or define the schema and automate this process.

Reviewer: ```suggestion
Read the Docs is notified of all changes to your git branches and git tags.
When you release a new version for your project,
you are likely also adding a version tag to your git repository.
These git events can be configured to build and publish your documentation automatically with :doc:`/automation-rules`.
If you use a versioning schema, you can configure it as part of the automation process.
```

Author: This is a really good simplifying change :+1:
